ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -273,23 +287,17 @@ func HTTPRouteToGateway(namespace, name, key string) ([]string, bool) {
 		if httpRouteStatus.Parents[statusIndex].ParentRef.Name != parentRef.Name {
 			continue
 		}
-		for statusIndex < len(httpRouteStatus.Parents) && (parentRef.SectionName != nil && *httpRouteStatus.Parents[statusIndex].ParentRef.SectionName != *parentRef.SectionName) {
```

Comment:
we do not require this condition? CC: @pkoshtavmware

Reply:
No @akshayhavile, we don't need this.